JSONObject.java

```diff
@@ -706,20 +706,17 @@ public long getLong(String key) throws JSONException {
     }
 
     /**
-     * Get the string associated with a key.
+     * Get the string representation of the value associated with a key.
      *
      * @param key
      *            A key string.
-     * @return A string which is the value.
+     * @return The string representation of the value.
      * @throws JSONException
-     *             if there is no string value for the key.
+     *             if there is no value for the key.
      */
     public String getString(String key) throws JSONException {
         Object object = this.get(key);
-        if (object instanceof String) {
-            return (String) object;
-        }
-        throw new JSONException("JSONObject[" + quote(key) + "] not a string.");
+        return object.toString();
```

Comment:
Just a note for other reviewers, the get(key) will throw an exception if the key or the value is null, so no null check is needed here.

I'm not sure if you want to make this more succinct though as
```java
return this.get(key).toString();
```

I'm personally not a fan of local variables that are only used once.